### DISCUSSION TITLE
What is the equivalent of getRules to get all rules that a plugin defines?

### DISCUSSION THREAD

**Discussion by gajus** [eslint/eslint#15210] [Q&A] [answered]
This is how I used to do it:

https://github.com/gajus/eslint-config-canonical/blob/474e0812c91c1f190b0ee5e432b507af5599ceb6/test/eslint-config-canonical/canonical.js#L19-L23

However, it seems this is no longer possible.

[Migration guidelines](https://eslint.org/docs/8.0.0/user-guide/migrating-to-8.0.0#remove-cliengine) do not mention how to get all enabled rules based on ESLint configuration:

> ※2 The CLIEngine#getRules() method had side effects and so was removed. If you were using CLIEngine#getRules() to retrieve meta information about rules based on linting results, use ESLint#getRulesMetaForResults() instead. If you were using CLIEngine#getRules() to retrieve all built-in rules, import builtinRules from eslint/use-at-your-own-risk for an unsupported API that allows access to internal rules.

**Comment by nzakas** [comment, marked as answer]
All plugins export a “rules” key. So you can pull the list of plugins from your config, then pull the rules directly off the plugin objects.

**Reply by gajus** [reply]
For others reference, here is an example implementation using Nicholas' suggestion.

https://github.com/gajus/eslint-config-canonical/blob/master/test/eslint-config-canonical/canonical.js
